### PR TITLE
[test] fix broken client tests

### PR DESCRIPTION
**Pull request by m4dz**
This PR include some client tests fixes, and (finally) enable client tests runs on CI. It also prepares `RequestsStore` spec.

### CODE REVIEW

File: client/app/stores/message_store.coffee [hunk 1]
```diff
@@ -49,7 +49,7 @@ class MessageStore extends Store
             message.date          ?= new Date().toISOString()
             message.createdAt     ?= message.date
             message.flags         ?= []
-            message.hasAttachments = message.attachments.length > 0
+            message.hasAttachments = message.attachments.size > 0
```
Comment (misstick) [inline client/app/stores/message_store.coffee]: @m4dz: a bug should appear according to `line48` : if no attachement this will crash.

**Comment by misstick** [comment]
here a solution for `RECEIVE_RAW_MESSAGES` test issue:

file: `client/stores/message_store.coffee`
````
    _updateMessages = (result={}, timestamp) ->
        {messages, conversationLength} = result

        # This prevent to override local updates
        # with older ones from server
        messages?.forEach (message) ->
            _saveMessage message, timestamp if message

        # ...
```

**Comment by misstick** [comment]
[feature] remove test on no-implemented feature
file: `client/test/message_store.spec.js`
`````
-    it('SEARCH_SUCCESS', () => {
+   it.skip('SEARCH_SUCCESS', () => {
+      // TODO: update this test when feature will be back
````

**Comment by misstick** [comment]
[fix] displayImage test
````
    it('SETTINGS_UPDATE_REQUEST', () => {
      const id1 = fixtures.message1.id;

      // Message must exist into messageStore
      assert.equal(messageStore.getByID(id1).get('id'), id1);
      assert.isUndefined(messageStore.getByID(id1).get('_displayImages'));

      // diaplayImage value has changed  
      dispatcher.dispatch({
        type: ActionTypes.SETTINGS_UPDATE_RESQUEST,
        value: { messageID: id1, displayImages: true },
      });
      assert.isTrue(messageStore.getByID(id1).get('_displayImages'));

      // diaplayImage value has changed
      dispatcher.dispatch({
        type: ActionTypes.SETTINGS_UPDATE_RESQUEST,
        value: { messageID: id1, displayImages: false },
      });
      assert.isFalse(messageStore.getByID(id1).get('_displayImages'));
    });
```

**Comment by misstick** [comment]
[test] fix `getConversation` typo

1/ fixtures have typo
````
-    mailboxIDs: ['inbox'],
+    mailboxIDs: { inbox: 1 },
```

**Comment by misstick** [comment]
[test] fix `getConversationLenth`
file: `client/test/message_store.spec.js`
`````
     it('getConversationLength', () => {
+      const id1 = fixtures.message10.id;
       const conversationId1 = fixtures.message10.conversationID;
+
+      // Message10 should exist into messageStore
+      assert.deepEqual(messageStore.getByID(id1).toObject(), fixtures.message10);
+
+      // Its length is 2
+      let length = messageStore.getConversationLength(conversationId1);
+      assert.equal(length, 1);
+
+      const id2 = fixtures.message11.id;
       const conversationId2 = fixtures.message11.conversationID;
-      let length = messageStore.getConversationLength(conversationId2);
+
+      // Message11 should exist into messageStore
+      assert.deepEqual(messageStore.getByID(id2).toObject(), fixtures.message11);
+
+      // Its length is 2
+      length = messageStore.getConversationLength(conversationId2);
       assert.equal(length, 2);
-      length = messageStore.getConversationLength(conversationId1);
-      assert.equal(length, 1);
+
+      // Empty conversation should return length: null
       length = messageStore.getConversationLength('c5');
-      assert.isUndefined(length);
+      assert.isNull(length);
`````

**Comment by m4dz** [comment]
@misstick in _fix flagSuccess_, `const updated = {messages: [message1, message2] }` will never ever work, as, if I understand properly http://github.com/m4dz/cozy-emails/blob/0ec836194b8f6d8f63232d413e38ee6c1e7ea60d/client/app/actions/router_action_creator.coffee#L233, there is no batch flaging: `updated` always contains only one `message`. Nope ?

**Pull request merged**